tests/select_point.rs

```diff
@@ -0,0 +1,359 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+//
+// Copyright (c) DUSK NETWORK. All rights reserved.
+
+use dusk_plonk::prelude::*;
+use rand::rngs::StdRng;
+use rand::SeedableRng;
+
+mod common;
+use common::{check_satisfied_circuit, check_unsatisfied_circuit, setup};
+
+#[test]
+fn component_select_point() {
+    pub struct TestCircuit {
+        bit: BlsScalar,
+        point_a: JubJubAffine,
+        point_b: JubJubAffine,
+        result: JubJubAffine,
+    }
+
+    impl TestCircuit {
+        pub fn new(
+            bit: BlsScalar,
+            point_a: JubJubAffine,
+            point_b: JubJubAffine,
+            result: JubJubAffine,
+        ) -> Self {
+            Self {
+                bit,
+                point_a,
+                point_b,
+                result,
+            }
+        }
+    }
+
+    impl Default for TestCircuit {
+        fn default() -> Self {
+            Self::new(
+                BlsScalar::zero(),
+                JubJubAffine::identity(),
+                JubJubAffine::identity(),
+                JubJubAffine::identity(),
+            )
+        }
+    }
+
+    impl Circuit for TestCircuit {
+        fn circuit<C>(&self, composer: &mut C) -> Result<(), Error>
+        where
+            C: Composer,
+        {
+            let w_bit = composer.append_witness(self.bit);
+            let w_point_a = composer.append_point(self.point_a);
+            let w_point_b = composer.append_point(self.point_b);
+            let w_result = composer.append_point(self.result);
+
+            let result_circuit =
+                composer.component_select_point(w_bit, w_point_a, w_point_b);
+
+            composer.assert_equal_point(w_result, result_circuit);
+
+            Ok(())
+        }
+    }
```

Comment:
Should we, for the sake of example completeness, constrait the `bit` to be 0 or 1?

Reply:
You're completely right!

Reply:
On second thought though: I actually added tests that pass even though `bit` is not 1 or 0 to specifically test that behavior. So I might opt to add a comment, that clarifies that any `component_select` should _always_ constrain `bit` _before_ the component is called.

Reply:
In any case: the `component_select*` tests were part of the last issue #732 so I'll go ahead and merge this one